mex.go

```diff
@@ -180,12 +179,16 @@ func (mexset *messageExchangeSet) newExchange(ctx context.Context, framePool Fra
 			mexset.log.Warnf("msg id %d used for both active mex %s and new mex %s",
 				mex.msgID, existingMex.msgType, mex.msgType)
 		}
+		mexset.mut.Unlock()
```

Comment:
nit: order this so it looks like:
```go
if [...] {
  [...]
} else {
  [...]
}

mexset.mut.Unlock()
return nil, errDuplicateMex
```

I want the `Unlock` as close to the return